librz/analysis/analysis.c

```diff
@@ -446,18 +446,24 @@ RZ_API void rz_analysis_purge(RzAnalysis *analysis) {
 	rz_analysis_purge_imports(analysis);
 }
 
-RZ_API int rz_analysis_archinfo(RzAnalysis *analysis, int query) {
-	rz_return_val_if_fail(analysis, -1);
+RZ_API int rz_analysis_archinfo(RzAnalysis *analysis, RzAnalysisInfoType query) {
+	rz_return_val_if_fail(analysis && query < RZ_ANALYSIS_ARCHINFO_ENUM_SIZE, -1);
+	if (!analysis->cur || !analysis->cur->archinfo) {
+		return -1;
+	}
+
+	int value = analysis->cur->archinfo(analysis, query);
 	switch (query) {
 	case RZ_ANALYSIS_ARCHINFO_MIN_OP_SIZE:
-	case RZ_ANALYSIS_ARCHINFO_MAX_OP_SIZE:
-	case RZ_ANALYSIS_ARCHINFO_ALIGN:
-		if (analysis->cur && analysis->cur->archinfo) {
-			return analysis->cur->archinfo(analysis, query);
-		}
-		break;
+		// Always consume at least 1 byte
+		return value > 0 ? value : 1;
+	case RZ_ANALYSIS_ARCHINFO_CAN_USE_POINTERS:
+		// When negative we assume the architecture does use them.
```

Comment:
I don't understand this... For example, amd29k returns true for `RZ_ANALYSIS_ARCHINFO_CAN_USE_POINTERS`, which is positive, why do you say negative is true?

Reply:
you can assume that  when a developer does not handle `RZ_ANALYSIS_ARCHINFO_CAN_USE_POINTERS` the return value can be `-1`, but we always want to return a boolean, thus i prefer to handle that case even tho -1 is "true"